src/Nethermind/Nethermind.Merge.AuRa.Test/ShutterCryptoTests.cs

```diff
@@ -0,0 +1,225 @@
+    //     "ebb6d602c2116e88cc88d14c4e1c97ace16c4479027707fb5dfdbacfece4fcb1",
+    //     "02ad644e5e3d095829d4f8a9b44d679887b922a78b783667522a12b18ad994f773ac06c3b47a4389efdbb9a6a6e97ff3701613d8e9e3a0d9defd520b353263bc73a15ee9b4729b59f6856bfa36bb1aff259d3df281749908cba45cdb8cba8555674ba8c0b3f617888d4e5fa647edff4f9888c42ef83537b804ee3ece3894bf8b608561ab85e78f093dc13de48a1091863b105e8f053dbe87efb72ad1aa115c4678"
+    // )]
+    // public void Can_encrypt_data(string msgHex, string identityPreimageHex, string eonKeyHex, string sigmaHex, string expectedHex)
+    // {
+    //     byte[] msg = Convert.FromHexString(msgHex);
+    //     byte[] expected = Convert.FromHexString(expectedHex);
+
+    //     G1 identity = ShutterCrypto.ComputeIdentity(Convert.FromHexString(identityPreimageHex));
+    //     G2 eonKey = new(Convert.FromHexString(eonKeyHex));
+    //     Bytes32 sigma = new(Convert.FromHexString(sigmaHex).AsSpan());
+
+    //     EncryptedMessage c = Encrypt(msg, identity, eonKey, sigma);
+
+    //     byte[] encoded = EncodeEncryptedMessage(c);
+    //     TestContext.WriteLine("encrypted msg: " + Convert.ToHexString(encoded));
+    //     Assert.That(encoded, Is.EqualTo(expected));
+    // }
+
+    // cryptotests decryption 4
+    // [Test]
+    // [TestCase(
+    //     "0290633abedfed746bd1a571f0b90da2fb66d6ce2f1ff36197095fdb6ac1a354694d01f2437a44a9d3a733a772ccc502e71769f538451cc5e048e649b77adda9c3adb3689d3869364d99151286f9ca5db57083cb5f37c6179b409e5ea1ecdff55a4d02ef4c0e48e9a35f0c818c15d5cb5166a29905ab68621a5042b42993e82d75a0d387431f41e354342f80c7d8079aceb3898357f0d6336eee3a6e057863116fe8bd47c38df1667709579bf3eedc8f3fbc7dc6cf099135eb364db97d551a29b1",
+    //     "89301819c6d24ef7555d822d6ad3543fdf28ecc149ba1c4359527c406fd700a75db6025d07010dc35a440da372b1318a",
+    //     "f12853fff4e9d2e997039e317f789e1684c4c8766cfce77a3153321601091079"
+    // )]
+    // public void Can_decrypt_data(string cipherTextHex, string decryptionKeyHex, string expectedHex)
+    // {
+    //     EncryptedMessage c = ShutterCrypto.DecodeEncryptedMessage(Convert.FromHexString(cipherTextHex));
+    //     G1 decryptionKey = new(Convert.FromHexString(decryptionKeyHex));
+
+    //     // recover sigma
+    //     GT p = new(decryptionKey, c.c1);
+    //     Bytes32 key = ShutterCrypto.Hash2(p);
+    //     Bytes32 sigma = ShutterCrypto.XorBlocks(c.c2, key);
+
+    //     // decrypt
+    //     IEnumerable<Bytes32> keys = ShutterCrypto.ComputeBlockKeys(sigma, c.c3.Count());
+    //     IEnumerable<Bytes32> decryptedBlocks = Enumerable.Zip(keys, c.c3, ShutterCrypto.XorBlocks);
+
+    //     byte[] decryptedMessage = ShutterCrypto.Decrypt(c, decryptionKey);
+    //     TestContext.WriteLine("decrypted msg: " + Convert.ToHexString(decryptedMessage));
+
+    //     Assert.That(decryptedMessage.SequenceEqual(Convert.FromHexString(expectedHex)));
+    // }
```

Comment:
Confused about commented tests

Reply:
Some of the crypto code is commented out until one of Shutter's components is updated. When the code is uncommented these tests are passing with some test vectors they supplied